### ISSUE TITLE
In POSIX sh, jobs flags beside -l and -p are undefined

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1429]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
jobs -s
jobs -r
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
The only acceptable flags under POSIX sh for `jobs` are `-l` and `-p` ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/jobs.html)). The `-s` and `-r` flags are [valid on Bash](https://www.gnu.org/software/bash/manual/html_node/Job-Control-Builtins.html#Job-Control-Builtins) to check for stopped/suspended jobs and running jobs, but invalid on, for instance, dash:
```sh
$ jobs -s
dash: 1: jobs: Illegal option -s
$ jobs -r
dash: 2: jobs: Illegal option -r
```

I would expect shellcheck to warn under SC2039: "In POSIX sh, jobs flags beside -l and -p are undefined".

If you agree with the issue, I can submit a pull request adding the check.
